railgun/test/integration/ingress_test.go

```diff
@@ -164,3 +222,146 @@ func TestHTTPSRedirect(t *testing.T) {
+
+	defer func() {
+		t.Logf("cleaning up the deployment %s", deployment.Name)
+		assert.NoError(t, cluster.Client().AppsV1().Deployments("default").Delete(ctx, deployment.Name, metav1.DeleteOptions{}))
+	}()
+
+	t.Logf("exposing deployment %s via service", deployment.Name)
+	service := k8sgen.NewServiceForDeployment(deployment, corev1.ServiceTypeLoadBalancer)
+	_, err = cluster.Client().CoreV1().Services("default").Create(ctx, service, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	defer func() {
+		t.Logf("cleaning up the service %s", service.Name)
+		assert.NoError(t, cluster.Client().CoreV1().Services("default").Delete(ctx, service.Name, metav1.DeleteOptions{}))
+	}()
+
+	t.Logf("creating an ingress for service %s without ingress.class %s", service.Name, ingressClass)
+	ingress := k8sgen.NewIngressForService("/httpbin", map[string]string{"konghq.com/strip-path": "true"}, service)
+	ingress.Spec.IngressClassName = kong.String(ingressClass)
+	ingress, err = cluster.Client().NetworkingV1().Ingresses("default").Create(ctx, ingress, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	defer func() {
+		t.Logf("ensuring that Ingress %s is cleaned up", ingress.Name)
+		if err := cluster.Client().NetworkingV1().Ingresses("default").Delete(ctx, ingress.Name, metav1.DeleteOptions{}); err != nil {
+			if !errors.IsNotFound(err) {
+				require.NoError(t, err)
+			}
+		}
+	}()
+
+	t.Logf("waiting for routes from Ingress %s to be operational", ingress.Name)
+	p := proxyReady()
+	assert.Eventually(t, func() bool {
+		resp, err := httpc.Get(fmt.Sprintf("%s/httpbin", p.ProxyURL.String()))
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", p.ProxyURL.String(), err)
+			return false
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusOK {
+			// now that the ingress backend is routable, make sure the contents we're getting back are what we expect
+			// Expected: "<title>httpbin.org</title>"
+			b := new(bytes.Buffer)
+			b.ReadFrom(resp.Body)
+			return strings.Contains(b.String(), "<title>httpbin.org</title>")
+		}
+		return false
+	}, ingressWait, waitTick)
+
+	t.Logf("removing the IngressClassName \"%s\" from ingress %s", ingressClass, ingress.Name)
+	ingress, err = cluster.Client().NetworkingV1().Ingresses("default").Get(ctx, ingress.Name, metav1.GetOptions{})
+	assert.NoError(t, err)
+	ingress.Spec.IngressClassName = nil
+	ingress, err = cluster.Client().NetworkingV1().Ingresses("default").Update(ctx, ingress, metav1.UpdateOptions{})
+	assert.NoError(t, err)
+
+	t.Logf("verifying that removing the IngressClassName \"%s\" from ingress %s causes routes to disconnect", ingressClass, ingress.Name)
+	assert.Eventually(t, func() bool {
+		resp, err := httpc.Get(fmt.Sprintf("%s/httpbin", p.ProxyURL.String()))
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", p.ProxyURL.String(), err)
+			return false
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusNotFound {
```

Comment:
I think that we should either make a 404 status code suffice (regardless of body content), or implement the body differ as a helper function - because this minute detail adds a lot of weight to the code of the test case.

Reply:
resolved by 4e9927f59ad2f4d4f09c358b70aea4d864a253b8 after some talk during our sync today :+1: